### PR TITLE
Feature/sem 3 g2

### PR DESCRIPTION
**Pull request by Vecc2424**
frontend for Access and Sharing of Scientific Articles [SEM-3]

### CODE REVIEW

**Comment by olmedo90** [comment]
image

It is according to the figma, continue with the same interest. 👍👍👍

File: package.json [hunk 1]
```diff
@@ -19,6 +19,8 @@
     "postcss": "8.4.30",
     "react": "18.2.0",
     "react-dom": "18.2.0",
+    "react-icons": "^4.12.0",
+    "react-syntax-highlighter": "^15.5.0",
```
Comment (Ditmar) [inline package.json:23]: Why you need to install `react-syntax-highlighter`

File: src/app/apiconection/api.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+export const fetchData = async (): Promise<Article[]> => {
+    try {
+        const response = await fetch('https://jsonplaceholder.typicode.com/comments');
```
Comment (Ditmar) [inline src/app/apiconection/api.ts:3]: ok , this is not good you need to separate the request in another file, you can create another directory called Async, and there you can create an API request service.
```
type RequestParams = {
    method: string;
    headers?: HeadersInit;
    body?: BodyInit | null;
};

type FetchOptions = {
    responseType: 'json' | 'blob';
};

async function fetchService<T>(
    url: string, 
    params: RequestParams, 
    options: FetchOptions
): Promise<T> {
    try {
        const response = await fetch(url, params);

        if (!response.ok) {
            throw new Error(`HTTP error! Status: ${response.status}`);
        }

        return options.responseType === 'json' ? await response.json() : await response.blob();
    } catch (error) {
        console.error('Fetching error:', error);
        throw error;
    }
}

async function getJsonData<T>(url: string): Promise<T> {
    return fetchService<T>(url, { method: 'GET' }, { responseType: 'json' });
}

// Example usage for Blob (file)
async function getFile(url: string): Promise<Blob> {
    return fetchService<Blob>(url, { method: 'GET' }, { responseType: 'blob' });
}

export { getJsonData, getFile };
```
And how to use it.
Suppose you have an API endpoint that returns user data in JSON format. You want to fetch this data and display it. Here's how you can use the getJsonData **function:**

```
import { getJsonData } from './fetchService'; 

interface UserData {
    id: number;
    name: string;
    email: string;
    // ... other user fields
}

async function fetchUserData() {
    try {
        const url = 'https://example.com/api/users'; 
        const userData = await getJsonData<UserData[]>(url);
        console.log('User Data:', userData);
        // Process and display the user data as needed
    } catch (error) {
        console.error('Error fetching user data:', error);
    }
}

// Call this function where you need to fetch the user data
fetchUserData();
```
please take this approach and refactor your code

File: src/app/page.tsx [hunk 3]
```diff
@@ -1,7 +1,8 @@
+import ArticulosCietificos from "@/components/ArticulosCientificos";
 export default function Home() {
   return (
     <main className="flex min-h-screen flex-col items-center justify-between p-24">
-      Main Data
+      <ArticulosCietificos/>
```
Comment (Ditmar) [inline src/app/page.tsx:5]: User english for the name of your components example ScientificArticles

File: src/app/apiconection/api.ts (hunk 2 above)
Comment (Ditmar) [inline src/app/apiconection/api.ts:3]: dont hard code the urls use the envs. process.env.SOME_URL

File: src/components/ArticulosCientificos/index.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  };
+
+  return (
+    <div className="flex min-h-screen items-top justify-between p-24">
+      <div className="w-3/4 mr-16 relative">
+        {articles.length > 0 && (
+          <>
+
+            <div className="slider-content">
+              <p className="article-recent">ARTÍCULOS MAS RECIENTES</p>
```
Comment (Ditmar) [inline src/components/ArticulosCientificos/index.tsx:46]: divide all this code in most small components

File: src/components/ArticulosCientificos/styles.scss [hunk 5]
```diff
@@ -0,0 +1,89 @@
+@import url('https://fonts.googleapis.com/css2?family=Roboto&display=swap');
```
Comment (Ditmar) [inline src/components/ArticulosCientificos/styles.scss:1]: read this article to improve the approach for importing the fonts https://nextjs.org/docs/pages/building-your-application/optimizing/fonts

**Review by Ditmar — CHANGES_REQUESTED** [review]

**Review by Ditmar — APPROVED** [review]
good job

**Review by Ditmar — APPROVED** [review]

**Review by Ditmar — APPROVED** [review]

**Pull request merged**